### PR TITLE
[Tensor] Unsigned Quantized Tensor

### PR DESCRIPTION
**Pull request by djeong20**
- Quantized tensor values are unsigned with zero points
- Layer context dequantize quantized tensor when request weight
- Template dequantize function

**Self evaluation:**
1. Build test:   [X]Passed [ ]Failed [ ]Skipped
2. Run test:     [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/tensor/tensor.cpp [hunk 1]
```diff
@@ -227,6 +241,31 @@ bool Tensor::operator==(const Tensor &rhs) const {
 #else
     throw std::invalid_argument("Error: enable-fp16 is not enabled");
 #endif
+  } else if (dim.getDataType() == ml::train::TensorDim::DataType::QINT8) {
+    const uint8_t *_data = getData<uint8_t>();
```
Comment (jijoongmoon) [inline nntrainer/tensor/tensor.cpp:245]: we might need to check the zero point and scale as well.

Comment (djeong20) [reply nntrainer/tensor/tensor.cpp:251]: applied.

**Review by Hyoungsukim — APPROVED** [review]
lgtm

**Review by songgot — APPROVED** [review]
LGTM

**Pull request merged**